checkov/common/runners/runner_registry.py

```diff
@@ -99,3 +114,74 @@ def enrich_report_with_guidelines(scan_report, guidelines):
         for record in scan_report.failed_checks + scan_report.passed_checks + scan_report.skipped_checks:
             if record.check_id in guidelines:
                 record.set_guideline(guidelines[record.check_id])
+
+    @staticmethod
+    def get_enriched_resources(report, repo_root):
+        parser = Parser()
+        tf_definitions = {}
+        parsing_errors = {}
+        Parser.parse_directory(
+            parser,
+            directory=repo_root,  # assume plan file is in the repo-root
+            out_definitions=tf_definitions,
+            out_parsing_errors=parsing_errors,
+        )
+
+        enriched_resources = defaultdict(dict)
```

Comment:
If the resource isn't found in the repo-root, it won't get enriched. When `enriched_resources` is used in the `enrich_plan_report` function to lookup what the resource enrichments are, the defaultdict prevents a keyerror if that resource can't be found.

Reply:
The reason to use `defaultdict` is correct, but your code won't raise any error, because of now using `enriched_resources.get(...)`
https://github.com/bridgecrewio/checkov/blob/9559952090aa517de3d49e5deb62a7134fc852b7/checkov/common/runners/runner_registry.py#L161-L162

Reply:
Ack. Makes sense.